compiler/src/compiler/mir_optimize/constant_folding.rs

```diff
@@ -233,6 +233,23 @@ impl Mir {
                     | Expression::TraceFoundFuzzableClosure { .. } => unreachable!(),
                 }
             }
+            BuiltinFunction::TextConcatenate => {
+                // TODO: Properly implement this optimization.
+                //       This is just a preliminary implementation to get use working
+                //       with the changes introduced by text interpolation.
```

Comment:
Unless I'm overlooking something, I don't think this TODO comment is necessary

Reply:
One possible improvement would be to reduce calls where one argument is an empty text to the other argument.

Reply:
Resolved by 3413ba6 and 60922cc